pkg/termination/termination.go

```diff
@@ -17,29 +17,60 @@ package termination
 
 import (
 	"encoding/json"
+	"io/ioutil"
 	"log"
 	"os"
 
 	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 	"go.uber.org/zap"
 )
 
+func Exists(name string) bool {
+	info, err := os.Stat(name)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
 func WriteMessage(logger *zap.SugaredLogger, path string, pro []v1alpha1.PipelineResourceResult) {
+	// if the file at path exists, concatenate the new values otherwise create it
+	exists := Exists(path)
+	if exists {
```

Comment:
`ioutil.ReadFile(path)` will return an error that `os.IsNotExists` can accept, and `os.OpenFile` will happily overwrite the file without it being deleted first. So I think we can remove `Exists(name)` and the call to `os.Remove(path)`, ending up with something like this:

```go
fileContents, err := ioutil.ReadFile(path)
if err == nil {
  if len(fileContents) != 0 {
    var existingEntries []v1alpha1.PipelineResourceResult
    if err := json.Unmarshal([]byte(fileContents), &existingEntries); err != nil {
      logger.Fatalf("Could not parse json message '%q' because of %w\n", fileContents, err)
    }
    pro = append(existingEntries, pro...)
  }
} else if ! os.IsNotExist(err) {
  logger.Fatalf("unexpected error reading %q: %v", path, err)
}
jsonOutput, err := json.Marshal(pro)
// ... and so on ...
```

Even this could be simplified - if `json.Unmarshal` returns an error I'd argue we simply print it and continue on as if the file didnt exist.  So we wouldn't need to worry about the `len(fileContents)` check either...

Reply:
ok done and pushed. I am still looking at other errors in other tests.